src/Microsoft.DotNet.Interactive.ExtensionLab/InspectExtension.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Html;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.ExtensionLab.Inspector;
+using Microsoft.DotNet.Interactive.Formatting;
+
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive.ExtensionLab
+{
+    public class InspectExtension : IKernelExtension
+    {
+        private const string INSPECT_COMMAND = "inspect";
+        public Task OnLoadAsync(Kernel kernel)
+        {
+            var inspect = new Command($"#!{INSPECT_COMMAND}", "Inspect the following code in the submission")
+            {
+                new Option<OptimizationLevel>(
+                        new [] {"-c", "--configuration"},
+                        getDefaultValue: () => OptimizationLevel.Debug,
+                        description: "Build configuration to use. Debug or Release."),
+                new Option<SourceCodeKind>(
+                        new [] {"-k", "--kind"},
+                        getDefaultValue: () => SourceCodeKind.Script,
+                        description: "Source code kind. Script or Regular."),
+            };
+
+            inspect.Handler = CommandHandler.Create((OptimizationLevel configuration, SourceCodeKind kind, Platform platform, KernelInvocationContext context) => Inspect(configuration, kind, platform, context));
+
+            kernel.AddDirective(inspect);
+
+
+            return Task.CompletedTask;
+        }
+
+        private void Inspect(OptimizationLevel configuration, SourceCodeKind kind, Platform platform, KernelInvocationContext context)
+        {
+            if (!(context.Command is SubmitCode))
+                return;
+
+            var command = context.Command as SubmitCode;
+
+            // TODO: Is there a proper way of cleaning up code from the magic commands?
```

Comment:
The `SubmitCode` command has an internal `LanguageNode` that might be useful.  Maybe we should make it public?

Reply:
Yeah, that will be useful, I think.